### ISSUE TITLE
Adding a new transaction with a date older than last date in security History window.

### ISSUE THREAD

**Issue by Derekkent23** [TheInfiniteKind/moneydance_open#22]
When I enter the transaction in my investment account and set the date back, the price in the price field has to be edited. Would it be possible to take a price from the Currency/Security History window if one exists for that date? See video to illustrate. 

https://user-images.githubusercontent.com/59656614/113899676-eff30f00-97c4-11eb-917b-a6eb1fab3684.mp4

**Comment by sreilly** [comment]
How would you like this to work specifically - as in would it be preferable that changing the date field in a not-yet-recorded transaction would automatically update the price field? Of course we'd have to be really careful not to auto-update the price field if it had been manually changed by the customer.

**Comment by yogi1967** [comment]
I don't think you should record a snapshot price until a txn is actually recorded.... and if the user has manually edited the txn price then that should take precedence (IMHO). I'm not actually sure how this enhancement would work.. Personally, if I've already set a price, and then change the date, I would still want the price I set in the txn to hold.. I guess it depends how you work. I can see Derek's view, but that's not how I work....? Just an opinion...

**Comment by sreilly** [comment]
Right, a snapshot price isn't recorded until the transaction is saved, and even then it isn't recorded if there is already a historical price on that date.

The issue here, if I'm understanding correctly, is the price that is automatically populated into the price field when entering a new transaction.

**Comment by yogi1967** [comment]
Ah... so if this is a brand new txn, then I guess that changing the date would search for a snapshot price and use that if it exists.... but once I edit / change the txn price field manually then it's fixed (but I can see how that would be problematic to program); Derek - how did you see it?

**Comment by Derekkent23** [comment]
To perform a Buy/Sell(Xfr) I click NEW TRANSACTION and the Date in the Date Field is highlighted so that the natural starting point. I set the Date I want to use for the transaction. I use TAB to move from field to field as well as tabbing through the fields I don’t fill in. Work flow is left to right which fell natural. 
So:
NEW TRANSACTION – Set highlighted Date – TAB – Set Action Buy/Sell(Xfr) – TAB Add Description if required - TAB - TAB - Select SECURITY (At this point if a price for that Security exists in the Security History window for the selected Date fill it in in the Price field, otherwise leave it blank waiting a manual entry) – TAB – Type number of SHARES. The user can then edit the PRICE field then ENTER or just accept the suggested PRICE by using ENTER. Before using ENTER Tab can be used to move to edit any other of the raining fields. The use of ENTER is when the data is saved.

**Comment by yogi1967** [comment]
I agree with Derek's suggestion on new txns.....

**Comment by NickWard999** [comment]
Derek's "way of working" is how I use the security screens today also.

**Comment by dwg1957** [comment]
I'm not sure of the value of this.

Typically your price history will show closing prices if a tool like Quote Loader is used to populate your history. I think it would be infrequent that you would be buying or selling at the closing price. As such I think it may make it too easier to just see and accept the price that is auto entered without checking that it is the correct price.

**Comment by NickWard999** [comment]
@dwg1957, I hear what you are saying, although, I have a use case scenario, where the mutual funds that I buy each month through payroll deductions, I manually enter each one, approx. 100 a month, due to the fund companies in ability to produce a download file of anything near a useable format (do not get me started on that subject).. So for the first round of entries that I make (multiple accounts and the MF company splits between employee and employer contributions) I enter the fund unit prices, there after each time I enter a new transaction for the same fund, different unit qty's the price of the fund on that day should populate into the transaction, regardless of account... being used.

**Comment by Derekkent23** [comment]
dwg, for me and I and sure I cant be alone on this the closing price provided by Quote Loader is nearly always the price used by my fund provider, so yes it’s a benefit to have a price loaded from the selected date in Security History window when performing a buy/sell(Xfr)transaction. If it’s wrong it can still be edited, so no change for those who need to edit. 

Let’s assume the price for all my Securities turn out not to match the price used by my fund provider. The suggestion is still a great time saver.

 I have the same set of Securities across a number of Investment Accounts. If necessary I can manually correct the Dated Price using Quote Loader or Security Price Entry or in the Security History window import a CSV file or import a CSV file using one of the other Security Price CSV file importer extensions by Mike Bray. If my suggestion is implemented then once the prices is amended using one of the above methods when I add transactions for the same Security across different Investment Accounts the correct price on the transaction date is entered automatically saying a lot of time.

**Comment by dwg1957** [comment]
Derek I can see where what you suggest works for funds since they generally only update their price at most once a day and I see many funds that update prices only once a week, I have some in fact. 

I was thinking of shares where the price floats during the day.

Moneydance essentially treats funds and shares in an identical fashion. Yet they do not work the same, and this is the key here I think, the differences have caused me some problems from day 1. I have made some suggestions in the past where funds need to be treated differently to shares, and this to me seems to be another place.

**Comment by Derekkent23** [comment]
My suggestion does not affect users who need to edit the price as part of a Buy/Sell (Xfr) transaction, they still can as now. Perhaps you need to post a separate issue related to how Stocks and Funds should be treated differently.

**Comment by sreilly** [comment]
Would this be solved with a keyboard shortcut and/or button to set the price field for the transaction from the security's price history?

**Comment by Derekkent23** [comment]
The process for entering a new investment transaction is:
1.	Enter the transaction Date.
2.	Follow the process as now until you enter the Security field and select a Security. At this point the Price field should automatically take up the Price in the associated Security History window for the entered transaction Date. If no price exists for that date the next oldest price should be used. This would make the transaction straight forward. 

However, having a keyboard shortcut and/or button to change the price from the Current Price would also work. Maybe easier to implement would be to have a selection in Preferences INVESTMENT TRANSACTIONS – USE TRANSACTION DATE PRICE INSTEAD OF CURRENT PRICE.

In both cases the user can still edit the Price field as now.
